okhttp/src/main/java/okhttp3/internal/http/Http1xStream.java

```diff
@@ -406,9 +397,8 @@ public FixedLengthSource(long length) throws IOException {
     @Override public void close() throws IOException {
       if (closed) return;
 
-      if (bytesRemaining != 0
-          && !Util.discard(this, DISCARD_STREAM_TIMEOUT_MILLIS, MILLISECONDS)) {
-        unexpectedEndOfInput();
+      if (bytesRemaining != 0 && !Util.discard(this, DISCARD_STREAM_TIMEOUT_MILLIS, MILLISECONDS)) {
+        endOfInput(false);
```

Comment:
Is there a test for that? Could we add one? Only because it seems prone to breakage as highly-underused behavior (the automatic exhausting, that is).